### PR TITLE
Fix IPGlobalProperties.DomainName to return string.Empty when domain name is not set

### LINKED ISSUE #104882: `IPGlobalProperties.DomainName` returns `"(none)"` on Linux
### Description

as per [Documentation](https://learn.microsoft.com/en-us/dotnet/api/system.net.networkinformation.ipglobalproperties.domainname?view=net-8.0), the property should either return the actual domain name or `string.Empty`.

### Reproduction Steps

- have a Linux install, where the domain is missing(?)
- use eg. `pwsh`
- run `[System.Net.NetworkInformation.IPGlobalProperties]::GetIPGlobalProperties().DomainName`

### Expected behavior

either return the actual domain name or `string.Empty`

### Actual behavior

returns `"(none)"`

image

### Regression?

(we're using C# on Linux for the first time here, so no idea)

### Known Workarounds

- `pwsh`
- `$h = [System.Net.DNS]::GetHostname()`
- `[System.Net.DNS]::GetHostEntry($h).HostName`
- returns equivalent to `BUILD-SERVER-LX.company.local`

But this approach might be unreliable if the DNS is not reachable, or perhaps misconfigured(?)

### Configuration

Local:
- WSL-Ubuntu
- `Linux MY-WORK-PC-123 4.4.0-19041-Microsoft #4355-Microsoft Thu Apr 12 17:37:00 PST 2024 x86_64 x86_64 x86_64 GNU/Linux`
-  `/etc/hosts` contains a line a la `127.0.1.1 MY-WORK-PC-123.company.local MY-WORK-PC-123` (actual names replaced)
- tests (via VS-WSL) are passing

Build Server:
- `Linux BUILD-SERVER-LX 5.15.0-113-generic #123-Ubuntu SMP Mon Jun 10 08:16:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
- no such hosts line
- tests fail, as the FQDN determined is `"BUILD-SERVER-LX.(none)"`, which was (with additional strings) passed to `new Uri(...)`

This issue was caught by our tests (net6.0), but `[Environment]::Version` (`pwsh`) shows 8.0.6 (ob both machines) and it experiences the same behavior (build server)

Not sure if C# gets the information from the hosts file, just something i noticed. WSL re-creates the file upon restart, and on the build server I currently do not have sufficient permissions to change the host file.

### Other information

_No response_

### PR DESCRIPTION
**Pull request by hootanht**
Fixes #104882

Fix `IPGlobalProperties.DomainName` to return `string.Empty` instead of `"(none)"` on Linux.

* **HostInformationPal.Unix.cs**
  - Check if the domain name is `"(none)"` in `GetDomainName()`.
  - Return `string.Empty` if the domain name is `"(none)"`.

* **Interop.GetDomainName.cs**
  - Handle the case where the domain name is not set in `GetDomainName()`.
  - Return `string.Empty` if the domain name is `"(none)"`.

* **IPGlobalPropertiesTest.cs**
  - Add a test to verify that `IPGlobalProperties.DomainName` returns `string.Empty` when the domain name is not set.

### CODE REVIEW

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs [hunk 1]
```diff
@@ -1,6 +1,3 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
```
Comment (liveans) [inline src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs:3]: Don't remove this please, this is required header.

File: src/libraries/Common/src/System/Net/NetworkInformation/HostInformationPal.Unix.cs [hunk 2]
```diff
@@ -1,6 +1,3 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
```
Comment (liveans) [inline src/libraries/Common/src/System/Net/NetworkInformation/HostInformationPal.Unix.cs:3]: Same as above this is required header, don't remove this please.

File: src/libraries/System.Net.NetworkInformation/tests/FunctionalTests/IPGlobalPropertiesTest.cs [hunk 3]
```diff
@@ -1,6 +1,3 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
```
Comment (liveans) [inline src/libraries/System.Net.NetworkInformation/tests/FunctionalTests/IPGlobalPropertiesTest.cs:3]: Don't remove this please.

File: src/libraries/Common/src/System/Net/NetworkInformation/HostInformationPal.Unix.cs [hunk 4]
```diff
@@ -12,7 +9,12 @@ public static string GetHostName()
 
         public static string GetDomainName()
         {
-            return Interop.Sys.GetDomainName();
+            string domainName = Interop.Sys.GetDomainName();
+            if (domainName == "(none)")
+            {
+                return string.Empty;
+            }
+            return domainName;
```
Comment (liveans) [inline src/libraries/Common/src/System/Net/NetworkInformation/HostInformationPal.Unix.cs:17]: I think, this is not necessary, issue is reproducing on Linux, so Interop change should be enough.

**Comment by liveans** [comment]
Thanks for the contribution! Generally, LGTM. I left couple of comments.

**Comment by hootanht** [comment]
Thanks' @liveans. I fix all the code review requests.

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs [hunk 5]
```diff
@@ -31,7 +31,13 @@ internal static unsafe string GetDomainName()
                 throw new InvalidOperationException($"{nameof(GetDomainName)}: {err}");
             }
 
-            return Marshal.PtrToStringUTF8((IntPtr)name)!;
+            string domainName = Marshal.PtrToStringUTF8((IntPtr)name)!;
+            if (domainName == "(none)")
```
Comment (stephentoub) [inline src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs:35]: In what situation / on what platforms does getdomainname return "(none)"?

Comment (wfurt) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs:35]: that feel like stringification of `null`

Comment (wfurt) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs:35]: I check I get the string `(none)` directly from `getdomainname` on my Ubuntu VM. (no NIS configured e.g. default) Return value is 0 e.g. success.
on macOS I get empty string. 

So this change tries to interpret result returned by the underlying OS function. I did not test if that changes if you change OS locale.

Comment (wfurt) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs:35]: It does not seems to depend on locale
```
furt@kamna:~/projects/newtorkInfo$ ~/dotnet-latest/dotnet run
Obnovení dokončeno (0,5s)
Používáte verzi Preview rozhraní .NET. Viz: https://aka.ms/dotnet-support-policy
  newtorkInfo úspěšné (0,2s) → bin/Debug/net9.0/newtorkInfo.dll

Sestavení úspěšné za 1,0s
Name kamna domain (none)
```

there is some explanation here:
https://access.redhat.com/solutions/352963

and  Kernel github

https://github.com/torvalds/linux/blob/e32cde8d2bd7d251a8f9b434143977ddf13dcec6/include/linux/uts.h#L17

**Review by wfurt — APPROVED** [review]
LGTM.  thanks @hootanht

**Pull request merged**